File: fapi/src/main/java/com/netcracker/lazarev/tms/model/Page.java
package com.netcracker.lazarev.tms.model;

import lombok.Data;

import java.util.List;

@Data
public class Page<T> {
    private Sort sort;
    private Pageable pageable;
    private List<T> content;
    private long totalElements;
    private int totalPages;
    private int size;
    private boolean last;
    private int number;
    private int numberOfElements;
    private boolean first;
    private boolean empty;
}

File: frontend/src/app/models/page.ts
import {Sort} from './sort';
import {Pageable} from './pageable';

export class Page<T> {
  sort: Sort;
  pageable: Pageable;
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
  numberOfElements: number;
  last: boolean;
  first: boolean;
  empty: boolean;
}
